pkg/oc/bootstrap/clusterup/components/persistent-volumes/setup_persistent_volumes.go

```diff
@@ -0,0 +1,250 @@
+  local dir="${1}"
+  if [[ ! -d "${dir}" ]]; then
+    mkdir -p "${dir}"
+  fi
+  if ! chcon -t svirt_sandbox_file_t "${dir}" &> /dev/null; then
+    echo "Not setting SELinux content for ${dir}"
+  fi
+  chmod 770 "${dir}"
+}
+
+function create_pv() {
+  local basedir="${1}"
+  local name="${2}"
+
+  setup_pv_dir "${basedir}/${name}"
+  if ! oc get pv "${name}" &> /dev/null; then 
+    generate_pv "${basedir}" "${name}" | oc create -f -
+  else
+    echo "persistentvolume ${name} already exists"
+  fi
+}
+
+basedir="%[2]s"
+setup_pv_dir "${basedir}/registry"
+
+for i in $(seq -f "%%04g" 1 %[1]d); do
+  create_pv "${basedir}" "pv${i}"
+done
+`
+
+type SetupPersistentVolumesOptions struct {
+	InstallContext componentinstall.Context
+}
+
+func (c *SetupPersistentVolumesOptions) Name() string {
+	return "persistent-volumes"
+}
+
+func (c *SetupPersistentVolumesOptions) Install(dockerClient dockerhelper.Interface, logdir string) error {
+	kclient, err := kubernetes.NewForConfig(c.InstallContext.ClusterAdminClientConfig())
+	if err != nil {
+		return err
+	}
+	securityClient, err := securityclient.NewForConfig(c.InstallContext.ClusterAdminClientConfig())
+	if err != nil {
+		return err
+	}
+	authorizationClient, err := authorizationtypedclient.NewForConfig(c.InstallContext.ClusterAdminClientConfig())
+	if err != nil {
+		return err
+	}
+	if err := ensurePVInstallerSA(authorizationClient, kclient, securityClient); err != nil {
+		return err
+	}
+
+	// TODO: Make the job idempotent and non-failing
+	_, err = kclient.Batch().Jobs(pvTargetNamespace).Get(pvSetupJobName, metav1.GetOptions{})
+	if err == nil {
+		return nil
+	}
+	if !kerrors.IsNotFound(err) {
+		return fmt.Errorf("error retrieving job to setup persistent volumes (%s/%s): %v", pvTargetNamespace, pvSetupJobName, err)
+	}
+	targetDir := path.Join(c.InstallContext.BaseDir(), "openshift.local.pv")
+
+	if _, err := os.Stat(path.Join(targetDir, pvIgnoreMarkerFile)); !os.IsNotExist(err) {
+		glog.Infof("Found %q marker file, skipping persistent volume setup", path.Join(targetDir, pvIgnoreMarkerFile))
+		return nil
+	}
+
+	setupJob := persistentStorageSetupJob(pvSetupJobName, targetDir, c.InstallContext.ClientImage(), pvCount)
+	if _, err = kclient.Batch().Jobs(pvTargetNamespace).Create(setupJob); err != nil {
+		return fmt.Errorf("cannot create job to setup persistent volumes (%s/%s): %v", pvTargetNamespace, pvSetupJobName, err)
+	}
+
+	return nil
+}
+
+func ensurePVInstallerSA(authorizationClient authorizationtypedclient.ClusterRoleBindingsGetter, kclient kubernetes.Interface, securityClient securityclient.Interface) error {
+	createSA := false
+	sa, err := kclient.Core().ServiceAccounts(pvTargetNamespace).Get(pvInstallerSA, metav1.GetOptions{})
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			return fmt.Errorf("error retrieving installer service account (%s/%s): %v", pvTargetNamespace, pvInstallerSA, err)
+		}
+		createSA = true
+	}
+
+	// Create installer SA
+	if createSA {
```

Comment:
:sob: srsly? Since you're touching it please fix it. Just move the body of this if into the previous one.